applications/dashboard/controllers/Api/InvitationsApiController.php

```diff
@@ -0,0 +1,203 @@
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+use Garden\Schema\Schema;
+use Garden\Web\Data;
+use Garden\Web\Exception\ClientException;
+use Garden\Web\Exception\NotFoundException;
+use Vanilla\Utility\CapitalCaseScheme;
+
+/**
+ * API Controller for the `/invitations` resource.
+ */
+class InvitationsApiController extends AbstractApiController {
+
+    /** @var CapitalCaseScheme */
+    private $caseScheme;
+
+    /** @var InvitationModel */
+    private $invitationModel;
+
+    /** @var UserModel */
+    private $userModel;
+
+    /**
+     * InvitationsApiController constructor.
+     *
+     * @param CapitalCaseScheme $caseScheme
+     * @param Gdn_Configuration $configuration
+     * @param InvitationModel $invitationModel
+     * @param UserModel $userModel
+     * @throws ClientException if the site is not configured for user invitations.
+     */
+    public function __construct(CapitalCaseScheme $caseScheme, GDN_Configuration $configuration, InvitationModel $invitationModel, UserModel $userModel) {
+        $registrationMethod = strtolower($configuration->get('Garden.Registration.Method'));
+        if ($registrationMethod !== 'invitation') {
+            throw new ClientException('The site is not configured for the invitation registration method.');
+        }
+
+        $this->caseScheme = $caseScheme;
+        $this->invitationModel = $invitationModel;
+        $this->userModel = $userModel;
+    }
+
+    /**
+     * Delete an invitation.
+     *
+     * @param int $id The invitation ID.
+     */
+    public function delete($id) {
+        $this->permission('Garden.SignIn.Allow');
+
+        $in = $this->idParamSchema()->setDescription('Delete an invitation.');
+        $out = $this->schema([], 'out');
+
+        $row = $this->invitationByID($id);
+
+        if ($row['InsertUserID'] !== $this->getSession()->UserID) {
+            $this->permission('Garden.Moderation.Manage');
+        }
+
+        $this->invitationModel->deleteID($id);
+    }
+
+    /**
+     * Get a schema instance comprised of all available invitation fields.
+     *
+     * @return Schema Returns a schema object.
+     */
+    public function fullSchema() {
+        static $schema;
+
+        if ($schema === null) {
+            $schema = Schema::parse([
+                'invitationID:i' => 'A unique numerical ID for the invitation.',
+                'email:s' => 'The email address associated with an invitation.',
+                'code:s' => 'An invitation code.',
+                'status' => [
+                    'description' => 'Current status for the invitation.',
+                    'enum' => ['accepted', 'pending']
+                ],
+                'insertUserID:i' => 'User who created the invitation.',
+                'dateInserted:dt' => 'When the invitation was created.',
+                'acceptedUser:o|n?' => $this->getUserFragmentSchema(),
+                'acceptedUserID:i|n' => 'User who accepted the invitation.',
```

Comment:
That's modifying the underlying structure of this resource. I agree it would be a nice bit of information to have, but I believe it's outside the scope of this update.

Reply:
Structural updates added in a separate PR: #6008